Sources/AST/Environment/Environment.swift

```diff
@@ -175,6 +176,28 @@ public struct Environment {
     return sourceSelf == target
   }
 
+  /// Whether two function signatures are compatible.
+  ///
+  /// - Parameters:
+  ///   - source: signature declaration of the function that the user is trying to use.
+  ///   - target: signature declaration of the function available in this scope.
+  ///   - enclosingType: Type identifier of type containing *source* function.
+  /// - Returns: Boolean indicating whether two function arguments are compatible.
```

Comment:
Returns should probably read whether two function *signatures* are compatible. I think it might also be worth to explain in more detail what "compatible" means, i.e. that the kinds of arguments have to be the same (or `Self`-compatible), including `inout` modifiers, etc.

Reply:
okie doke

Reply:
<img width="517" alt="screenshot 2018-10-26 at 22 10 12" src="https://user-images.githubusercontent.com/2105177/47592652-13ff6480-d96c-11e8-936e-7171b03f73eb.png">
done